development/extensions/index.md

```diff
@@ -24,51 +24,125 @@ One of the primary aims of BlueOS is to be a platform that is readily extendable
+{{ easy_image(src="anatomy", width=600) }}
+
+At heart, a BlueOS Extension is some functionality (optionally with a web interface) packaged into a [Docker Image](../overview/#docker), and combined with some metadata that allows it to be found, shared, and managed. Extensions may interface with existing services provided by BlueOS (or other Extensions), and some Extensions may create persistent logs and/or make use of data or files provided by the user.
+
+Once installed, an Extension Package can be run as a [Docker Container](../overview/#docker), which normally occurs automatically when the vehicle turns on, but can also be manually disabled/enabled via the Extensions Manager. When running, Extensions can have [custom permissions assigned](#metadata-dockerfile), which can limit resource-usage and/or allow access to parts of the host computer's hardware.
+
+#### Where Does the Code Run?
+
+To integrate well within BlueOS, extensions typically include a [backend service](#backend-services) that runs on the onboard computer (in the vehicle), which provides some functionality, and is often accessed by a [frontend interface](#frontend-interfaces) that runs in the control station computer's browser.
+
+{{ easy_image(src="frontend-backend", width=600, center=true) }}
+
+A common usage process would follow:
+1. Vehicle turns on, and backend service initialises
+2. A user requests the frontend interface, from the browser in their control station computer
+3. Backend serves a web interface to the browser
+4. User interacts with the interface, via some kind of input
+    - button clicks, text submissions, joystick events, etc.
+5. Frontend makes API calls to its backend service to facilitate the desired interaction(s)
+6. Backend receives an API call, processes it, and responds appropriately
+    - processing typically involves logging the request, accessing some hardware, calculating something, and/or requesting something from another service
+7. Frontend receives the response from the backend, and updates the interface accordingly
+    - the frontend may also do some processing, such as decoding an image to display to the user
+8. Repeat from step 4. until further interaction is no longer required
+
+##### Backend Services
+- Communicate with each other (via [APIs](../../advanced-usage/#available-services))
+- Typically run continuously while the vehicle is on
+- Have access to the vehicle computer hardware, including connected devices and the filesystem
+- May have internet access, if the vehicle computer does
+- May serve a graphical web-interface that's accessible via the frontend
+    - Extensions without a graphical interface could include device drivers and/or something with fully pre-determined functionality (that doesn't require user input)
+
+##### Frontend Interfaces
```

Comment:
It would be nice to add a note for simple frontend only projects.
Maybe we could link to one of our examples and add a note about https://github.com/TheWaWaR/simple-http-server